// sdksio-monoova-sdk/dist/cjs/models/disbursement1.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.disbursement1Schema = void 0;
var schema_1 = require("../schema");
var disbursementMethod1Enum_1 = require("./disbursementMethod1Enum");
var toBPayDetails_1 = require("./toBPayDetails");
var toDirectCreditDetails_1 = require("./toDirectCreditDetails");
var toNppCreditBankAccountDetails_1 = require("./toNppCreditBankAccountDetails");
var toNppCreditPayIdDetails_1 = require("./toNppCreditPayIdDetails");
exports.disbursement1Schema = (0, schema_1.object)({
    disbursementMethod: [
        'disbursementMethod',
        (0, schema_1.optional)(disbursementMethod1Enum_1.disbursementMethod1EnumSchema),
    ],
    description: ['description', (0, schema_1.optional)((0, schema_1.string)())],
    toMAccount: ['toMAccount', (0, schema_1.optional)((0, schema_1.string)())],
    amount: ['amount', (0, schema_1.optional)((0, schema_1.number)())],
    toMWallet: ['toMWallet', (0, schema_1.optional)((0, schema_1.string)())],
    toDirectCreditDetails: [
        'toDirectCreditDetails',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return toDirectCreditDetails_1.toDirectCreditDetailsSchema; })),
    ],
    lodgementReference: ['lodgementReference', (0, schema_1.optional)((0, schema_1.string)())],
    toToken: ['toToken', (0, schema_1.optional)((0, schema_1.string)())],
    toBPayDetails: ['toBPayDetails', (0, schema_1.optional)((0, schema_1.lazy)(function () { return toBPayDetails_1.toBPayDetailsSchema; }))],
    toNppCreditBankAccountDetails: [
        'toNppCreditBankAccountDetails',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return toNppCreditBankAccountDetails_1.toNppCreditBankAccountDetailsSchema; })),
    ],
    toNppCreditPayIdDetails: [
        'toNppCreditPayIdDetails',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return toNppCreditPayIdDetails_1.toNppCreditPayIdDetailsSchema; })),
    ]
});
