// sdksio-monoova-sdk/dist/cjs/models/dTOBPAYBillerV1.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.dTOBPAYBillerV1Schema = void 0;
var schema_1 = require("../schema");
exports.dTOBPAYBillerV1Schema = (0, schema_1.object)({
    billerCode: ['billerCode', (0, schema_1.optional)((0, schema_1.string)())],
    billerLongName: ['billerLongName', (0, schema_1.optional)((0, schema_1.string)())],
    billerShortName: ['billerShortName', (0, schema_1.optional)((0, schema_1.string)())],
    acceptedPaymentMethods: ['acceptedPaymentMethods', (0, schema_1.optional)((0, schema_1.string)())],
    activationDate: ['activationDate', (0, schema_1.optional)((0, schema_1.string)())],
    deactivationDate: ['deactivationDate', (0, schema_1.optional)((0, schema_1.string)())],
    minPaymentAmount: ['minPaymentAmount', (0, schema_1.optional)((0, schema_1.number)())],
    maxPaymentAmount: ['maxPaymentAmount', (0, schema_1.optional)((0, schema_1.number)())],
    crnValidationRuleName: ['crnValidationRuleName', (0, schema_1.optional)((0, schema_1.string)())],
    checkDigitRuleName: ['checkDigitRuleName', (0, schema_1.optional)((0, schema_1.string)())],
    lengthMask: ['lengthMask', (0, schema_1.optional)((0, schema_1.string)())],
    fixedDigits: ['fixedDigits', (0, schema_1.optional)((0, schema_1.string)())],
    isVariableCrn: ['isVariableCrn', (0, schema_1.optional)((0, schema_1.boolean)())]
});
