// sdksio-monoova-sdk/dist/cjs/models/dTOBPAYBillersResponseV1.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.dTOBPAYBillersResponseV1Schema = void 0;
var schema_1 = require("../schema");
var dTOBPAYBillerV1_1 = require("./dTOBPAYBillerV1");
exports.dTOBPAYBillersResponseV1Schema = (0, schema_1.object)({
    durationMs: ['durationMs', (0, schema_1.optional)((0, schema_1.bigint)())],
    status: ['status', (0, schema_1.optional)((0, schema_1.string)())],
    statusDescription: ['statusDescription', (0, schema_1.optional)((0, schema_1.string)())],
    search: ['search', (0, schema_1.optional)((0, schema_1.string)())],
    totalCount: ['totalCount', (0, schema_1.optional)((0, schema_1.number)())],
    skip: ['skip', (0, schema_1.optional)((0, schema_1.number)())],
    take: ['take', (0, schema_1.optional)((0, schema_1.number)())],
    billers: ['billers', (0, schema_1.optional)((0, schema_1.array)((0, schema_1.lazy)(function () { return dTOBPAYBillerV1_1.dTOBPAYBillerV1Schema; })))]
});
