// sdksio-monoova-sdk/dist/cjs/models/dTOGenericPaymentGatewayTransactionStatusResponseV2.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.dTOGenericPaymentGatewayTransactionStatusResponseV2Schema = void 0;
var schema_1 = require("../schema");
var transactionStatusEnum_1 = require("./transactionStatusEnum");
exports.dTOGenericPaymentGatewayTransactionStatusResponseV2Schema = (0, schema_1.object)({
    durationMs: ['durationMs', (0, schema_1.optional)((0, schema_1.bigint)())],
    status: ['status', (0, schema_1.optional)((0, schema_1.string)())],
    statusDescription: ['statusDescription', (0, schema_1.optional)((0, schema_1.string)())],
    completedDate: ['completedDate', (0, schema_1.optional)((0, schema_1.string)())],
    creditCardPaymentStatus: ['creditCardPaymentStatus', (0, schema_1.optional)((0, schema_1.string)())],
    dishonouredDate: ['dishonouredDate', (0, schema_1.optional)((0, schema_1.string)())],
    expectedClearanceDateForFunds: [
        'expectedClearanceDateForFunds',
        (0, schema_1.optional)((0, schema_1.string)()),
    ],
    fundsClearedDate: ['fundsClearedDate', (0, schema_1.optional)((0, schema_1.string)())],
    transactionStatus: [
        'transactionStatus',
        (0, schema_1.optional)(transactionStatusEnum_1.transactionStatusEnumSchema),
    ],
    uniqueReference: ['uniqueReference', (0, schema_1.optional)((0, schema_1.string)())]
});
