// sdksio-monoova-sdk/dist/cjs/models/dTOGenericPaymentGenericPaymentSettlementItemV1.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.dTOGenericPaymentGenericPaymentSettlementItemV1Schema = void 0;
var schema_1 = require("../schema");
exports.dTOGenericPaymentGenericPaymentSettlementItemV1Schema = (0, schema_1.object)({
    uniqueReferenced: ['uniqueReferenced', (0, schema_1.optional)((0, schema_1.string)())],
    transactionId: ['transactionId', (0, schema_1.optional)((0, schema_1.number)())],
    disbursementAmount: ['disbursementAmount', (0, schema_1.optional)((0, schema_1.number)())],
    disbursementMethod: ['disbursementMethod', (0, schema_1.optional)((0, schema_1.string)())],
    feeAmountExclGst: ['feeAmountExclGst', (0, schema_1.optional)((0, schema_1.number)())],
    feeAmountGstComponent: ['feeAmountGstComponent', (0, schema_1.optional)((0, schema_1.number)())],
    feeAmountInclGst: ['feeAmountInclGst', (0, schema_1.optional)((0, schema_1.number)())]
});
