// sdksio-monoova-sdk/dist/cjs/models/dTOGenericPaymentGenericPaymentStatementItemV1.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.dTOGenericPaymentGenericPaymentStatementItemV1Schema = void 0;
var schema_1 = require("../schema");
exports.dTOGenericPaymentGenericPaymentStatementItemV1Schema = (0, schema_1.object)({
    uniqueReferenced: ['uniqueReferenced', (0, schema_1.optional)((0, schema_1.string)())],
    transactionId: ['transactionId', (0, schema_1.optional)((0, schema_1.number)())],
    dateTime: ['dateTime', (0, schema_1.optional)((0, schema_1.string)())],
    status: ['status', (0, schema_1.optional)((0, schema_1.string)())],
    totalAmount: ['totalAmount', (0, schema_1.optional)((0, schema_1.number)())],
    feeAmountExcludingGst: ['feeAmountExcludingGst', (0, schema_1.optional)((0, schema_1.number)())],
    feeAmountGstComponent: ['feeAmountGstComponent', (0, schema_1.optional)((0, schema_1.number)())],
    feeAmountIncludingGst: ['feeAmountIncludingGst', (0, schema_1.optional)((0, schema_1.number)())]
});
