// sdksio-monoova-sdk/dist/cjs/models/dTOGenericPaymentSettingsResponseV1.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.dTOGenericPaymentSettingsResponseV1Schema = void 0;
var schema_1 = require("../schema");
var dTOGenericPaymentDisbursementFeeV1_1 = require("./dTOGenericPaymentDisbursementFeeV1");
var dTOGenericPaymentLoadFeeV1_1 = require("./dTOGenericPaymentLoadFeeV1");
exports.dTOGenericPaymentSettingsResponseV1Schema = (0, schema_1.object)({
    durationMs: ['durationMs', (0, schema_1.optional)((0, schema_1.bigint)())],
    status: ['status', (0, schema_1.optional)((0, schema_1.string)())],
    statusDescription: ['statusDescription', (0, schema_1.optional)((0, schema_1.string)())],
    disbursementFees: [
        'disbursementFees',
        (0, schema_1.optional)((0, schema_1.array)((0, schema_1.lazy)(function () { return dTOGenericPaymentDisbursementFeeV1_1.dTOGenericPaymentDisbursementFeeV1Schema; }))),
    ],
    loadFees: [
        'loadFees',
        (0, schema_1.optional)((0, schema_1.array)((0, schema_1.lazy)(function () { return dTOGenericPaymentLoadFeeV1_1.dTOGenericPaymentLoadFeeV1Schema; }))),
    ],
    canDebitCreditCards: ['canDebitCreditCards', (0, schema_1.optional)((0, schema_1.boolean)())],
    canPayBpay: ['canPayBpay', (0, schema_1.optional)((0, schema_1.boolean)())],
    canImpersonate: ['canImpersonate', (0, schema_1.optional)((0, schema_1.boolean)())],
    canDirectDebit: ['canDirectDebit', (0, schema_1.optional)((0, schema_1.boolean)())],
    canDirectCredit: ['canDirectCredit', (0, schema_1.optional)((0, schema_1.boolean)())],
    canCreateMAccounts: ['canCreateMAccounts', (0, schema_1.optional)((0, schema_1.boolean)())],
    canCreateMWallets: ['canCreateMWallets', (0, schema_1.optional)((0, schema_1.boolean)())],
    canAccessUserDatabase: ['canAccessUserDatabase', (0, schema_1.optional)((0, schema_1.boolean)())],
    isIssuer: ['isIssuer', (0, schema_1.optional)((0, schema_1.boolean)())],
    issuerMAccountNumber: ['issuerMAccountNumber', (0, schema_1.optional)((0, schema_1.string)())],
    monthlyFeeExGst: ['monthlyFeeExGst', (0, schema_1.optional)((0, schema_1.number)())],
    clearingMAccountNumber: ['clearingMAccountNumber', (0, schema_1.optional)((0, schema_1.string)())],
    feeMAccountNumber: ['feeMAccountNumber', (0, schema_1.optional)((0, schema_1.string)())],
    feeMAccountMonthlyFeeExGst: [
        'feeMAccountMonthlyFeeExGst',
        (0, schema_1.optional)((0, schema_1.number)()),
    ],
    childMAccountSetUpFeeExGst: [
        'childMAccountSetUpFeeExGst',
        (0, schema_1.optional)((0, schema_1.number)()),
    ],
    childMAccountMonthlyFeeExGst: [
        'childMAccountMonthlyFeeExGst',
        (0, schema_1.optional)((0, schema_1.number)()),
    ],
    requiresClearedFundsOnly: ['requiresClearedFundsOnly', (0, schema_1.optional)((0, schema_1.boolean)())]
});
