// sdksio-monoova-sdk/dist/cjs/models/dTOGenericPaymentGatewayTransactionStatus.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.dTOGenericPaymentGatewayTransactionStatusSchema = void 0;
var schema_1 = require("../schema");
exports.dTOGenericPaymentGatewayTransactionStatusSchema = (0, schema_1.object)({
    completedDate: ['completedDate', (0, schema_1.optional)((0, schema_1.string)())],
    uniqueReference: ['uniqueReference', (0, schema_1.optional)((0, schema_1.string)())],
    statusDescription: ['statusDescription', (0, schema_1.optional)((0, schema_1.string)())],
    transactionStatus: ['transactionStatus', (0, schema_1.optional)((0, schema_1.string)())],
    dishonouredDate: ['dishonouredDate', (0, schema_1.optional)((0, schema_1.string)())],
    creditCardPaymentStatus: ['creditCardPaymentStatus', (0, schema_1.optional)((0, schema_1.unknown)())],
    waitingOnClearedFundsDate: [
        'waitingOnClearedFundsDate',
        (0, schema_1.optional)((0, schema_1.string)()),
    ]
});
