// sdksio-monoova-sdk/dist/cjs/models/dTOMWalletDetails.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.dTOMWalletDetailsSchema = void 0;
var schema_1 = require("../schema");
var dTOMWalletFinancials_1 = require("./dTOMWalletFinancials");
var dTOMWalletOptionItem_1 = require("./dTOMWalletOptionItem");
exports.dTOMWalletDetailsSchema = (0, schema_1.object)({
    createdDateTime: ['createdDateTime', (0, schema_1.optional)((0, schema_1.string)())],
    accountNumber: ['accountNumber', (0, schema_1.optional)((0, schema_1.string)())],
    name: ['name', (0, schema_1.optional)((0, schema_1.string)())],
    nickName: ['nickName', (0, schema_1.optional)((0, schema_1.string)())],
    email: ['email', (0, schema_1.optional)((0, schema_1.string)())],
    mobile: ['mobile', (0, schema_1.optional)((0, schema_1.string)())],
    lastTransactionDate: ['lastTransactionDate', (0, schema_1.optional)((0, schema_1.string)())],
    contactPhone: ['contactPhone', (0, schema_1.optional)((0, schema_1.string)())],
    authenticationEmail: ['authenticationEmail', (0, schema_1.optional)((0, schema_1.string)())],
    authenticationMobile: ['authenticationMobile', (0, schema_1.optional)((0, schema_1.string)())],
    authenticationLandLine: ['authenticationLandLine', (0, schema_1.optional)((0, schema_1.string)())],
    registeredByMAccountNumber: [
        'registeredByMAccountNumber',
        (0, schema_1.optional)((0, schema_1.string)()),
    ],
    isOnHold: ['isOnHold', (0, schema_1.optional)((0, schema_1.boolean)())],
    isClosed: ['isClosed', (0, schema_1.optional)((0, schema_1.boolean)())],
    isStolen: ['isStolen', (0, schema_1.optional)((0, schema_1.boolean)())],
    isRegistered: ['isRegistered', (0, schema_1.optional)((0, schema_1.boolean)())],
    financials: ['financials', (0, schema_1.optional)((0, schema_1.lazy)(function () { return dTOMWalletFinancials_1.dTOMWalletFinancialsSchema; }))],
    options: ['options', (0, schema_1.optional)((0, schema_1.array)((0, schema_1.lazy)(function () { return dTOMWalletOptionItem_1.dTOMWalletOptionItemSchema; })))]
});
