// sdksio-monoova-sdk/dist/cjs/models/dTOMAccountDetails.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.dTOMAccountDetailsSchema = void 0;
var schema_1 = require("../schema");
var dTOMAccountFinancials_1 = require("./dTOMAccountFinancials");
var dTOOptionItem_1 = require("./dTOOptionItem");
exports.dTOMAccountDetailsSchema = (0, schema_1.object)({
    accountNumber: ['accountNumber', (0, schema_1.optional)((0, schema_1.bigint)())],
    name: ['name', (0, schema_1.optional)((0, schema_1.string)())],
    abn: ['abn', (0, schema_1.optional)((0, schema_1.string)())],
    contactName: ['contactName', (0, schema_1.optional)((0, schema_1.string)())],
    contactNumber: ['contactNumber', (0, schema_1.optional)((0, schema_1.string)())],
    email: ['email', (0, schema_1.optional)((0, schema_1.string)())],
    addressLine1: ['addressLine1', (0, schema_1.optional)((0, schema_1.string)())],
    addressLine2: ['addressLine2', (0, schema_1.optional)((0, schema_1.string)())],
    suburb: ['suburb', (0, schema_1.optional)((0, schema_1.string)())],
    state: ['state', (0, schema_1.optional)((0, schema_1.string)())],
    postCode: ['postCode', (0, schema_1.optional)((0, schema_1.string)())],
    bsb: ['bsb', (0, schema_1.optional)((0, schema_1.string)())],
    bankAccountNumber: ['bankAccountNumber', (0, schema_1.optional)((0, schema_1.number)())],
    bankAccountTitle: ['bankAccountTitle', (0, schema_1.optional)((0, schema_1.string)())],
    allowDuplicates: ['allowDuplicates', (0, schema_1.optional)((0, schema_1.boolean)())],
    feeAccount: ['feeAccount', (0, schema_1.optional)((0, schema_1.number)())],
    financials: ['financials', (0, schema_1.optional)((0, schema_1.lazy)(function () { return dTOMAccountFinancials_1.dTOMAccountFinancialsSchema; }))],
    options: ['options', (0, schema_1.optional)((0, schema_1.array)((0, schema_1.lazy)(function () { return dTOOptionItem_1.dTOOptionItemSchema; })))]
});
