// sdksio-monoova-sdk/dist/cjs/models/dTOMWalletTransactionsResponseV1.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.dTOMWalletTransactionsResponseV1Schema = void 0;
var schema_1 = require("../schema");
var dTOMWalletTransactionLineItem_1 = require("./dTOMWalletTransactionLineItem");
exports.dTOMWalletTransactionsResponseV1Schema = (0, schema_1.object)({
    durationMs: ['durationMs', (0, schema_1.optional)((0, schema_1.bigint)())],
    status: ['status', (0, schema_1.optional)((0, schema_1.string)())],
    statusDescription: ['statusDescription', (0, schema_1.optional)((0, schema_1.string)())],
    openingBalance: ['openingBalance', (0, schema_1.optional)((0, schema_1.number)())],
    closingBalance: ['closingBalance', (0, schema_1.optional)((0, schema_1.number)())],
    totalDebits: ['totalDebits', (0, schema_1.optional)((0, schema_1.number)())],
    totalCredits: ['totalCredits', (0, schema_1.optional)((0, schema_1.number)())],
    items: [
        'items',
        (0, schema_1.optional)((0, schema_1.array)((0, schema_1.lazy)(function () { return dTOMWalletTransactionLineItem_1.dTOMWalletTransactionLineItemSchema; }))),
    ]
});
