// sdksio-monoova-sdk/dist/cjs/models/dTOMWalletTransactionLineItem.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.dTOMWalletTransactionLineItemSchema = void 0;
var schema_1 = require("../schema");
exports.dTOMWalletTransactionLineItemSchema = (0, schema_1.object)({
    rowOrder: ['rowOrder', (0, schema_1.optional)((0, schema_1.number)())],
    mPaymentsId: ['mPaymentsId', (0, schema_1.optional)((0, schema_1.number)())],
    transactionId: ['transactionId', (0, schema_1.optional)((0, schema_1.number)())],
    dateTime: ['dateTime', (0, schema_1.optional)((0, schema_1.string)())],
    mAccountName: ['mAccountName', (0, schema_1.optional)((0, schema_1.string)())],
    transactionType: ['transactionType', (0, schema_1.optional)((0, schema_1.string)())],
    subTransactionType: ['subTransactionType', (0, schema_1.optional)((0, schema_1.string)())],
    credit: ['credit', (0, schema_1.optional)((0, schema_1.number)())],
    debit: ['debit', (0, schema_1.optional)((0, schema_1.number)())],
    runningBalance: ['runningBalance', (0, schema_1.optional)((0, schema_1.number)())],
    description: ['description', (0, schema_1.optional)((0, schema_1.string)())]
});
