// sdksio-monoova-sdk/dist/cjs/models/dTOMAccountTransactionLineItem.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.dTOMAccountTransactionLineItemSchema = void 0;
var schema_1 = require("../schema");
exports.dTOMAccountTransactionLineItemSchema = (0, schema_1.object)({
    credit: ['credit', (0, schema_1.optional)((0, schema_1.number)())],
    dateTime: ['dateTime', (0, schema_1.optional)((0, schema_1.string)())],
    debit: ['debit', (0, schema_1.optional)((0, schema_1.number)())],
    description: ['description', (0, schema_1.optional)((0, schema_1.string)())],
    destinationMAccountCardNumber: [
        'destinationMAccountCardNumber',
        (0, schema_1.optional)((0, schema_1.string)()),
    ],
    isWaitingForFundsToClear: ['isWaitingForFundsToClear', (0, schema_1.optional)((0, schema_1.boolean)())],
    mPaymentsId: ['mPaymentsId', (0, schema_1.optional)((0, schema_1.number)())],
    mWalletAccountNumber: ['mWalletAccountNumber', (0, schema_1.optional)((0, schema_1.string)())],
    rowOrder: ['rowOrder', (0, schema_1.optional)((0, schema_1.number)())],
    runningBalance: ['runningBalance', (0, schema_1.optional)((0, schema_1.number)())],
    sourceMAccountCardNumber: ['sourceMAccountCardNumber', (0, schema_1.optional)((0, schema_1.string)())],
    subTransactionType: ['subTransactionType', (0, schema_1.optional)((0, schema_1.string)())],
    transactionDescription: ['transactionDescription', (0, schema_1.optional)((0, schema_1.string)())],
    transactionId: ['transactionId', (0, schema_1.optional)((0, schema_1.number)())],
    transactionType: ['transactionType', (0, schema_1.optional)((0, schema_1.string)())],
    uniqueReference: ['uniqueReference', (0, schema_1.optional)((0, schema_1.string)())]
});
