// sdksio-monoova-sdk/dist/cjs/models/dTOMAccountFinancials.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.dTOMAccountFinancialsSchema = void 0;
var schema_1 = require("../schema");
var dTOMAccountClearedFundsAccount_1 = require("./dTOMAccountClearedFundsAccount");
exports.dTOMAccountFinancialsSchema = (0, schema_1.object)({
    accountNumber: ['accountNumber', (0, schema_1.optional)((0, schema_1.number)())],
    actualBalance: ['actualBalance', (0, schema_1.optional)((0, schema_1.number)())],
    clearedFundsOnly: ['clearedFundsOnly', (0, schema_1.optional)((0, schema_1.boolean)())],
    creditLimit: ['creditLimit', (0, schema_1.optional)((0, schema_1.number)())],
    feeAccountActualBalance: ['feeAccountActualBalance', (0, schema_1.optional)((0, schema_1.number)())],
    feeAccountNumber: ['feeAccountNumber', (0, schema_1.optional)((0, schema_1.number)())],
    hasSeparateFeeAccount: ['hasSeparateFeeAccount', (0, schema_1.optional)((0, schema_1.boolean)())],
    lowerCreditLimit: ['lowerCreditLimit', (0, schema_1.optional)((0, schema_1.number)())],
    minimumSettlementAmount: ['minimumSettlementAmount', (0, schema_1.optional)((0, schema_1.number)())],
    settlementFrequency: ['settlementFrequency', (0, schema_1.optional)((0, schema_1.string)())],
    upperCreditLimit: ['upperCreditLimit', (0, schema_1.optional)((0, schema_1.number)())],
    isFeeAccount: ['isFeeAccount', (0, schema_1.optional)((0, schema_1.boolean)())],
    isParentAccount: ['isParentAccount', (0, schema_1.optional)((0, schema_1.boolean)())],
    isParentClearedFunds: ['isParentClearedFunds', (0, schema_1.optional)((0, schema_1.boolean)())],
    clearedFundsAccountFinancials: [
        'clearedFundsAccountFinancials',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return dTOMAccountClearedFundsAccount_1.dTOMAccountClearedFundsAccountSchema; })),
    ]
});
