// sdksio-monoova-sdk/dist/cjs/models/dTOToolsBsbValidateResponseV1.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.dTOToolsBsbValidateResponseV1Schema = void 0;
var schema_1 = require("../schema");
exports.dTOToolsBsbValidateResponseV1Schema = (0, schema_1.object)({
    durationMs: ['durationMs', (0, schema_1.optional)((0, schema_1.bigint)())],
    status: ['status', (0, schema_1.optional)((0, schema_1.string)())],
    statusDescription: ['statusDescription', (0, schema_1.optional)((0, schema_1.string)())],
    bsb: ['bsb', (0, schema_1.optional)((0, schema_1.string)())],
    closed: ['closed', (0, schema_1.optional)((0, schema_1.boolean)())],
    address: ['address', (0, schema_1.optional)((0, schema_1.string)())],
    city: ['city', (0, schema_1.optional)((0, schema_1.string)())],
    state: ['state', (0, schema_1.optional)((0, schema_1.string)())],
    postCode: ['postCode', (0, schema_1.optional)((0, schema_1.string)())],
    bankCode: ['bankCode', (0, schema_1.optional)((0, schema_1.string)())]
});
