// sdksio-monoova-sdk/dist/cjs/models/dTOToolsAbnValidateResponseV1.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.dTOToolsAbnValidateResponseV1Schema = void 0;
var schema_1 = require("../schema");
exports.dTOToolsAbnValidateResponseV1Schema = (0, schema_1.object)({
    durationMs: ['durationMs', (0, schema_1.optional)((0, schema_1.bigint)())],
    status: ['status', (0, schema_1.optional)((0, schema_1.string)())],
    statusDescription: ['statusDescription', (0, schema_1.optional)((0, schema_1.string)())],
    abn: ['abn', (0, schema_1.optional)((0, schema_1.string)())],
    lastUpdatedDate: ['lastUpdatedDate', (0, schema_1.optional)((0, schema_1.string)())],
    isAbnValid: ['isAbnValid', (0, schema_1.optional)((0, schema_1.boolean)())],
    isCharity: ['isCharity', (0, schema_1.optional)((0, schema_1.boolean)())],
    entityTypeCode: ['entityTypeCode', (0, schema_1.optional)((0, schema_1.string)())],
    entityTypeDescription: ['entityTypeDescription', (0, schema_1.optional)((0, schema_1.string)())],
    organisationName: ['organisationName', (0, schema_1.optional)((0, schema_1.string)())],
    legalName: ['legalName', (0, schema_1.optional)((0, schema_1.string)())],
    organisationNameFromDate: ['organisationNameFromDate', (0, schema_1.optional)((0, schema_1.string)())],
    businessPhysicalAddressStateCode: [
        'businessPhysicalAddressStateCode',
        (0, schema_1.optional)((0, schema_1.string)()),
    ],
    businessPhysicalAddressPostCode: [
        'businessPhysicalAddressPostCode',
        (0, schema_1.optional)((0, schema_1.string)()),
    ],
    businessPhysicalAddressEffectiveFromDate: [
        'businessPhysicalAddressEffectiveFromDate',
        (0, schema_1.optional)((0, schema_1.string)()),
    ]
});
