// sdksio-monoova-sdk/dist/cjs/models/dTOVerifyAccountDetails.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.dTOVerifyAccountDetailsSchema = void 0;
var schema_1 = require("../schema");
exports.dTOVerifyAccountDetailsSchema = (0, schema_1.object)({
    token: ['token', (0, schema_1.optional)((0, schema_1.string)())],
    created: ['created', (0, schema_1.optional)((0, schema_1.string)())],
    updated: ['updated', (0, schema_1.optional)((0, schema_1.string)())],
    lastSystemBSBValidation: ['lastSystemBSBValidation', (0, schema_1.optional)((0, schema_1.string)())],
    comment: ['comment', (0, schema_1.optional)((0, schema_1.string)())],
    bsb: ['bsb', (0, schema_1.optional)((0, schema_1.string)())],
    bankAccountNumber: ['bankAccountNumber', (0, schema_1.optional)((0, schema_1.number)())],
    bankAccountTitle: ['bankAccountTitle', (0, schema_1.optional)((0, schema_1.string)())],
    remitter: ['remitter', (0, schema_1.optional)((0, schema_1.string)())],
    isBlacklisted: ['isBlacklisted', (0, schema_1.optional)((0, schema_1.boolean)())],
    hasDDAuthority: ['hasDDAuthority', (0, schema_1.optional)((0, schema_1.boolean)())],
    ddAuthorityChangedTimestamp: [
        'ddAuthorityChangedTimestamp',
        (0, schema_1.optional)((0, schema_1.string)()),
    ],
    ddAuthorityChangedBy: ['ddAuthorityChangedBy', (0, schema_1.optional)((0, schema_1.string)())],
    verificationIdentifier: ['verificationIdentifier', (0, schema_1.optional)((0, schema_1.string)())],
    verificationStatus: ['verificationStatus', (0, schema_1.optional)((0, schema_1.string)())],
    verificationSignonMerchant: [
        'verificationSignonMerchant',
        (0, schema_1.optional)((0, schema_1.string)()),
    ],
    verificationRetryCount: ['verificationRetryCount', (0, schema_1.optional)((0, schema_1.number)())],
    verificationRetyCountBeforeLockout: [
        'verificationRetyCountBeforeLockout',
        (0, schema_1.optional)((0, schema_1.number)()),
    ],
    verificationLastRetryTimestamp: [
        'verificationLastRetryTimestamp',
        (0, schema_1.optional)((0, schema_1.string)()),
    ],
    verificationLastRetryLockoutMinutes: [
        'verificationLastRetryLockoutMinutes',
        (0, schema_1.optional)((0, schema_1.number)()),
    ],
    verificationDEJobId: ['verificationDEJobId', (0, schema_1.optional)((0, schema_1.number)())],
    verificationRootId: ['verificationRootId', (0, schema_1.optional)((0, schema_1.number)())]
});
