// sdksio-monoova-sdk/dist/cjs/models/dTOVerifyAccountInitiateResponseV2.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.dTOVerifyAccountInitiateResponseV2Schema = void 0;
var schema_1 = require("../schema");
exports.dTOVerifyAccountInitiateResponseV2Schema = (0, schema_1.object)({
    durationMs: ['durationMs', (0, schema_1.optional)((0, schema_1.bigint)())],
    status: ['status', (0, schema_1.optional)((0, schema_1.string)())],
    statusDescription: ['statusDescription', (0, schema_1.optional)((0, schema_1.string)())],
    token: ['token', (0, schema_1.optional)((0, schema_1.string)())],
    feeAmountExGst: ['feeAmountExGst', (0, schema_1.optional)((0, schema_1.number)())],
    feeAmountIncGst: ['feeAmountIncGst', (0, schema_1.optional)((0, schema_1.number)())],
    feeAmountGstComp: ['feeAmountGstComp', (0, schema_1.optional)((0, schema_1.number)())],
    ownerName: ['ownerName', (0, schema_1.optional)((0, schema_1.string)())]
});
