// sdksio-monoova-sdk/dist/cjs/models/eventPayToreceivePaymentWebhook.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.eventPayToreceivePaymentWebhookSchema = void 0;
var schema_1 = require("../schema");
exports.eventPayToreceivePaymentWebhookSchema = (0, schema_1.object)({
    transactionId: ['TransactionId', (0, schema_1.optional)((0, schema_1.string)())],
    dateTime: ['DateTime', (0, schema_1.optional)((0, schema_1.string)())],
    remitterName: ['RemitterName', (0, schema_1.optional)((0, schema_1.string)())],
    amount: ['Amount', (0, schema_1.optional)((0, schema_1.string)())],
    accountName: ['AccountName', (0, schema_1.optional)((0, schema_1.string)())],
    accountNumber: ['AccountNumber', (0, schema_1.optional)((0, schema_1.string)())],
    bsb: ['Bsb', (0, schema_1.optional)((0, schema_1.string)())],
    paymentDescription: ['PaymentDescription', (0, schema_1.optional)((0, schema_1.string)())],
    payId: ['payId', (0, schema_1.optional)((0, schema_1.string)())],
    payIdName: ['payIdName', (0, schema_1.optional)((0, schema_1.string)())],
    sourceBsb: ['SourceBsb', (0, schema_1.optional)((0, schema_1.string)())],
    sourceAccountNumber: ['SourceAccountNumber', (0, schema_1.optional)((0, schema_1.string)())],
    sourceAccountName: ['SourceAccountName', (0, schema_1.optional)((0, schema_1.string)())],
    endToEndId: ['EndToEndId', (0, schema_1.optional)((0, schema_1.string)())],
    categoryPurposeCode: ['CategoryPurposeCode', (0, schema_1.optional)((0, schema_1.string)())],
    creditorReferenceInformation: [
        'CreditorReferenceInformation',
        (0, schema_1.optional)((0, schema_1.string)()),
    ],
    uSINumber: ['USINumber', (0, schema_1.optional)((0, schema_1.string)())],
    uSIScheme: ['USIScheme', (0, schema_1.optional)((0, schema_1.string)())],
    ultimateCreditorName: ['UltimateCreditorName', (0, schema_1.optional)((0, schema_1.string)())],
    mmsId: ['mmsId', (0, schema_1.optional)((0, schema_1.string)())],
    initiatingPartyName: ['InitiatingPartyName', (0, schema_1.optional)((0, schema_1.string)())]
});
