// sdksio-monoova-sdk/dist/cjs/models/eventInboundRtgsImtStatus.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.eventInboundRtgsImtStatusSchema = void 0;
var schema_1 = require("../schema");
exports.eventInboundRtgsImtStatusSchema = (0, schema_1.object)({
    transactionId: ['TransactionId', (0, schema_1.optional)((0, schema_1.string)())],
    accountNumber: ['AccountNumber', (0, schema_1.optional)((0, schema_1.string)())],
    bsb: ['Bsb', (0, schema_1.optional)((0, schema_1.string)())],
    accountName: ['AccountName', (0, schema_1.optional)((0, schema_1.string)())],
    transactionReferenceNumber: [
        'TransactionReferenceNumber',
        (0, schema_1.optional)((0, schema_1.string)()),
    ],
    amount: ['Amount', (0, schema_1.optional)((0, schema_1.number)())],
    status: ['Status', (0, schema_1.optional)((0, schema_1.string)())],
    lodgementRef: ['LodgementRef', (0, schema_1.optional)((0, schema_1.string)())],
    date: ['Date', (0, schema_1.optional)((0, schema_1.string)())],
    sourceDetails: ['SourceDetails', (0, schema_1.optional)((0, schema_1.string)())],
    processedDateTime: ['ProcessedDateTime', (0, schema_1.optional)((0, schema_1.string)())],
    rejectReasonDescription: ['RejectReasonDescription', (0, schema_1.optional)((0, schema_1.string)())],
    sendingOrganisation: ['SendingOrganisation', (0, schema_1.optional)((0, schema_1.string)())],
    receivingOrganisation: ['ReceivingOrganisation', (0, schema_1.optional)((0, schema_1.string)())],
    orderingInstitution: ['OrderingInstitution', (0, schema_1.optional)((0, schema_1.string)())],
    orderingAddress: ['OrderingAddress', (0, schema_1.optional)((0, schema_1.string)())]
});
