// sdksio-monoova-sdk/dist/cjs/models/eventNppPaymentStatusWebhook.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.eventNppPaymentStatusWebhookSchema = void 0;
var schema_1 = require("../schema");
var payIdType1Enum_1 = require("./payIdType1Enum");
exports.eventNppPaymentStatusWebhookSchema = (0, schema_1.object)({
    transactionDate: ['TransactionDate', (0, schema_1.optional)((0, schema_1.string)())],
    uniqueReference: ['UniqueReference', (0, schema_1.optional)((0, schema_1.string)())],
    transactionId: ['TransactionId', (0, schema_1.optional)((0, schema_1.number)())],
    status: ['Status', (0, schema_1.optional)((0, schema_1.string)())],
    amount: ['Amount', (0, schema_1.optional)((0, schema_1.number)())],
    bsb: ['Bsb', (0, schema_1.optional)((0, schema_1.string)())],
    accountNumber: ['AccountNumber', (0, schema_1.optional)((0, schema_1.string)())],
    accountName: ['AccountName', (0, schema_1.optional)((0, schema_1.string)())],
    payId: ['PayId', (0, schema_1.optional)((0, schema_1.string)())],
    payIdType: ['PayIdType', (0, schema_1.optional)(payIdType1Enum_1.payIdType1EnumSchema)],
    endToEndId: ['EndToEndId', (0, schema_1.optional)((0, schema_1.string)())],
    remitterName: ['RemitterName', (0, schema_1.optional)((0, schema_1.string)())],
    lodgementRef: ['LodgementRef', (0, schema_1.optional)((0, schema_1.string)())],
    rejectionDateTime: ['RejectionDateTime', (0, schema_1.optional)((0, schema_1.string)())],
    rejectionReasonDescription: [
        'RejectionReasonDescription',
        (0, schema_1.optional)((0, schema_1.string)()),
    ],
    rejectionTransactionId: ['RejectionTransactionId', (0, schema_1.optional)((0, schema_1.number)())]
});
