// sdksio-monoova-sdk/dist/cjs/models/eventNameEnum.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.eventNameEnumSchema = exports.EventNameEnum = void 0;
var schema_1 = require("../schema");
/**
 * Enum for EventNameEnum
 */
var EventNameEnum;
(function (EventNameEnum) {
    EventNameEnum["NppReturn"] = "NppReturn";
    EventNameEnum["NppPaymentStatus"] = "NppPaymentStatus";
    EventNameEnum["PayToReceivePayment"] = "PayToReceivePayment";
    EventNameEnum["NPPReceivePayment"] = "NPPReceivePayment";
    EventNameEnum["InboundDirectCredit"] = "InboundDirectCredit";
    EventNameEnum["InboundDirectDebit"] = "InboundDirectDebit";
    EventNameEnum["DirectEntryDishonour"] = "DirectEntryDishonour";
    EventNameEnum["PendingInboundRtgsImt"] = "PendingInboundRtgsImt";
    EventNameEnum["InboundRtgsImtStatus"] = "InboundRtgsImtStatus";
})(EventNameEnum = exports.EventNameEnum || (exports.EventNameEnum = {}));
/**
 * Schema for EventNameEnum
 */
exports.eventNameEnumSchema = (0, schema_1.stringEnum)(EventNameEnum);
