// sdksio-monoova-sdk/dist/cjs/models/eventNppReturnWebhook.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.eventNppReturnWebhookSchema = void 0;
var schema_1 = require("../schema");
var payIdType1Enum_1 = require("./payIdType1Enum");
exports.eventNppReturnWebhookSchema = (0, schema_1.object)({
    returnDate: ['ReturnDate', (0, schema_1.optional)((0, schema_1.string)())],
    amount: ['Amount', (0, schema_1.optional)((0, schema_1.number)())],
    bsb: ['Bsb', (0, schema_1.optional)((0, schema_1.string)())],
    accountNumber: ['AccountNumber', (0, schema_1.optional)((0, schema_1.string)())],
    accountName: ['AccountName', (0, schema_1.optional)((0, schema_1.string)())],
    payId: ['payId', (0, schema_1.optional)((0, schema_1.string)())],
    payIdType: ['payIdType', (0, schema_1.optional)(payIdType1Enum_1.payIdType1EnumSchema)],
    endToEndId: ['EndToEndId', (0, schema_1.optional)((0, schema_1.string)())],
    returnReason: ['ReturnReason', (0, schema_1.optional)((0, schema_1.string)())],
    transactionDate: ['TransactionDate', (0, schema_1.optional)((0, schema_1.string)())],
    transactionId: ['TransactionId', (0, schema_1.optional)((0, schema_1.number)())],
    uniqueReference: ['UniqueReference', (0, schema_1.optional)((0, schema_1.string)())],
    remitterName: ['RemitterName', (0, schema_1.optional)((0, schema_1.string)())],
    lodgementRef: ['LodgementRef', (0, schema_1.optional)((0, schema_1.string)())],
    returnTransactionId: ['ReturnTransactionId', (0, schema_1.optional)((0, schema_1.number)())]
});
