// sdksio-monoova-sdk/dist/cjs/models/eventDirectEntryDishonoursWebhook.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.eventDirectEntryDishonoursWebhookSchema = void 0;
var schema_1 = require("../schema");
var bankDetails_1 = require("./bankDetails");
exports.eventDirectEntryDishonoursWebhookSchema = (0, schema_1.object)({
    returnDate: ['ReturnDate', (0, schema_1.optional)((0, schema_1.string)())],
    amount: ['Amount', (0, schema_1.optional)((0, schema_1.number)())],
    bankDetails: ['BankDetails', (0, schema_1.optional)((0, schema_1.lazy)(function () { return bankDetails_1.bankDetailsSchema; }))],
    type: ['Type', (0, schema_1.optional)((0, schema_1.string)())],
    returnReason: ['ReturnReason', (0, schema_1.optional)((0, schema_1.string)())],
    transactionDate: ['TransactionDate', (0, schema_1.optional)((0, schema_1.string)())],
    originalTransactionId: ['OriginalTransactionId', (0, schema_1.optional)((0, schema_1.number)())],
    transactionReference: ['TransactionReference', (0, schema_1.optional)((0, schema_1.string)())]
});
