// sdksio-monoova-sdk/dist/cjs/models/inboundDirectCreditWebhookDirectCreditDetails.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.inboundDirectCreditWebhookDirectCreditDetailsSchema = void 0;
var schema_1 = require("../schema");
exports.inboundDirectCreditWebhookDirectCreditDetailsSchema = (0, schema_1.object)({
    transactionId: ['TransactionId', (0, schema_1.optional)((0, schema_1.string)())],
    batchId: ['BatchId', (0, schema_1.optional)((0, schema_1.string)())],
    dateTime: ['DateTime', (0, schema_1.optional)((0, schema_1.string)())],
    bsb: ['Bsb', (0, schema_1.optional)((0, schema_1.string)())],
    accountNumber: ['AccountNumber', (0, schema_1.optional)((0, schema_1.number)())],
    accountName: ['AccountName', (0, schema_1.optional)((0, schema_1.string)())],
    transactionCode: ['TransactionCode', (0, schema_1.optional)((0, schema_1.string)())],
    amount: ['Amount', (0, schema_1.optional)((0, schema_1.number)())],
    lodgementRef: ['LodgementRef', (0, schema_1.optional)((0, schema_1.string)())],
    remitterName: ['RemitterName', (0, schema_1.optional)((0, schema_1.string)())],
    nameOfUserSupplyingFile: ['NameOfUserSupplyingFile', (0, schema_1.optional)((0, schema_1.string)())],
    numberOfUserSupplyingFile: [
        'NumberOfUserSupplyingFile',
        (0, schema_1.optional)((0, schema_1.string)()),
    ],
    descriptionOfEntriesOnFile: [
        'DescriptionOfEntriesOnFile',
        (0, schema_1.optional)((0, schema_1.string)()),
    ],
    indicator: ['Indicator', (0, schema_1.optional)((0, schema_1.string)())],
    withholdingTaxAmount: ['WithholdingTaxAmount', (0, schema_1.optional)((0, schema_1.number)())],
    sourceBsb: ['SourceBsb', (0, schema_1.optional)((0, schema_1.string)())],
    sourceAccountNumber: ['SourceAccountNumber', (0, schema_1.optional)((0, schema_1.string)())]
});
