// sdksio-monoova-sdk/dist/cjs/models/mWalletCreate.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.mWalletCreateSchema = void 0;
var schema_1 = require("../schema");
var mobileForSms_1 = require("./mobileForSms");
exports.mWalletCreateSchema = (0, schema_1.object)({
    identifier: ['identifier', (0, schema_1.string)()],
    pin: ['pin', (0, schema_1.number)()],
    name: ['name', (0, schema_1.string)()],
    nickName: ['nickName', (0, schema_1.optional)((0, schema_1.string)())],
    dateOfBirth: ['dateOfBirth', (0, schema_1.string)()],
    phone: ['phone', (0, schema_1.optional)((0, schema_1.number)())],
    email: ['email', (0, schema_1.optional)((0, schema_1.string)())],
    allowDuplicates: ['allowDuplicates', (0, schema_1.optional)((0, schema_1.boolean)())],
    abn: ['abn', (0, schema_1.optional)((0, schema_1.string)())],
    options: ['options', (0, schema_1.optional)((0, schema_1.array)((0, schema_1.lazy)(function () { return mobileForSms_1.mobileForSmsSchema; })))]
});
