// sdksio-monoova-sdk/dist/cjs/models/notificationResponse.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.notificationResponseSchema = void 0;
var schema_1 = require("../schema");
var eventName1Enum_1 = require("./eventName1Enum");
var status1Enum_1 = require("./status1Enum");
exports.notificationResponseSchema = (0, schema_1.object)({
    dateTime: ['dateTime', (0, schema_1.optional)((0, schema_1.string)())],
    eventName: ['eventName', (0, schema_1.optional)(eventName1Enum_1.eventName1EnumSchema)],
    lastResponseCode: ['lastResponseCode', (0, schema_1.optional)((0, schema_1.number)())],
    lastRetryDateTime: ['lastRetryDateTime', (0, schema_1.optional)((0, schema_1.string)())],
    retryCount: ['retryCount', (0, schema_1.optional)((0, schema_1.number)())],
    status: ['status', (0, schema_1.optional)(status1Enum_1.status1EnumSchema)],
    url: ['url', (0, schema_1.optional)((0, schema_1.string)())],
    webhookId: ['webhookId', (0, schema_1.optional)((0, schema_1.number)())]
});
