// sdksio-monoova-sdk/dist/cjs/models/paymentSourceDirectDebit.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.paymentSourceDirectDebitSchema = void 0;
var schema_1 = require("../schema");
var directDebit_1 = require("./directDebit");
var disbursement1_1 = require("./disbursement1");
exports.paymentSourceDirectDebitSchema = (0, schema_1.object)({
    directDebit: ['directDebit', (0, schema_1.optional)((0, schema_1.lazy)(function () { return directDebit_1.directDebitSchema; }))],
    description: ['description', (0, schema_1.optional)((0, schema_1.string)())],
    lodgementReference: ['lodgementReference', (0, schema_1.optional)((0, schema_1.string)())],
    disbursements: [
        'disbursements',
        (0, schema_1.optional)((0, schema_1.array)((0, schema_1.lazy)(function () { return disbursement1_1.disbursement1Schema; }))),
    ],
    uniqueReference: ['uniqueReference', (0, schema_1.optional)((0, schema_1.string)())],
    printUniqueReferenceOnStatement: [
        'printUniqueReferenceOnStatement',
        (0, schema_1.optional)((0, schema_1.boolean)()),
    ],
    totalAmount: ['totalAmount', (0, schema_1.optional)((0, schema_1.number)())],
    paymentSource: ['paymentSource', (0, schema_1.optional)((0, schema_1.literal)('directDebit'))]
});
