// sdksio-monoova-sdk/dist/cjs/models/paymentSourceMWallet.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.paymentSourceMWalletSchema = void 0;
var schema_1 = require("../schema");
var disbursement1_1 = require("./disbursement1");
var mWallet1_1 = require("./mWallet1");
exports.paymentSourceMWalletSchema = (0, schema_1.object)({
    mWallet: ['mWallet', (0, schema_1.optional)((0, schema_1.lazy)(function () { return mWallet1_1.mWallet1Schema; }))],
    description: ['description', (0, schema_1.optional)((0, schema_1.string)())],
    disbursements: [
        'disbursements',
        (0, schema_1.optional)((0, schema_1.array)((0, schema_1.lazy)(function () { return disbursement1_1.disbursement1Schema; }))),
    ],
    uniqueReference: ['uniqueReference', (0, schema_1.optional)((0, schema_1.string)())],
    printUniqueReferenceOnStatement: [
        'printUniqueReferenceOnStatement',
        (0, schema_1.optional)((0, schema_1.boolean)()),
    ],
    totalAmount: ['totalAmount', (0, schema_1.optional)((0, schema_1.number)())],
    paymentSource: ['paymentSource', (0, schema_1.optional)((0, schema_1.literal)('mWallet'))]
});
