// sdksio-monoova-sdk/dist/cjs/models/paymentSourceMAccount.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.paymentSourceMAccountSchema = void 0;
var schema_1 = require("../schema");
var disbursement_1 = require("./disbursement");
var mAccount2_1 = require("./mAccount2");
exports.paymentSourceMAccountSchema = (0, schema_1.object)({
    mAccount: ['mAccount', (0, schema_1.optional)((0, schema_1.lazy)(function () { return mAccount2_1.mAccount2Schema; }))],
    description: ['description', (0, schema_1.optional)((0, schema_1.string)())],
    disbursements: [
        'disbursements',
        (0, schema_1.optional)((0, schema_1.array)((0, schema_1.lazy)(function () { return disbursement_1.disbursementSchema; }))),
    ],
    uniqueReference: ['uniqueReference', (0, schema_1.optional)((0, schema_1.string)())],
    printUniqueReferenceOnStatement: [
        'printUniqueReferenceOnStatement',
        (0, schema_1.optional)((0, schema_1.boolean)()),
    ],
    totalAmount: ['totalAmount', (0, schema_1.optional)((0, schema_1.number)())],
    paymentSource: ['paymentSource', (0, schema_1.optional)((0, schema_1.literal)('mAccount'))]
});
