// sdksio-monoova-sdk/dist/cjs/models/receivablesProcessDirectDebitResponseV1.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.receivablesProcessDirectDebitResponseV1Schema = void 0;
var schema_1 = require("../schema");
exports.receivablesProcessDirectDebitResponseV1Schema = (0, schema_1.object)({
    durationMs: ['durationMs', (0, schema_1.optional)((0, schema_1.bigint)())],
    status: ['status', (0, schema_1.optional)((0, schema_1.string)())],
    statusDescription: ['statusDescription', (0, schema_1.optional)((0, schema_1.string)())],
    uniqueReferenced: ['uniqueReferenced', (0, schema_1.optional)((0, schema_1.string)())],
    transactionId: ['transactionId', (0, schema_1.optional)((0, schema_1.number)())],
    feeAmountExcludingGst: ['feeAmountExcludingGst', (0, schema_1.optional)((0, schema_1.number)())],
    feeAmountGstComponent: ['feeAmountGstComponent', (0, schema_1.optional)((0, schema_1.number)())],
    feeAmountIncludingGst: ['feeAmountIncludingGst', (0, schema_1.optional)((0, schema_1.number)())]
});
