// sdksio-monoova-sdk/dist/cjs/models/receivablesEnquiryPayIdResponseV1.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.receivablesEnquiryPayIdResponseV1Schema = void 0;
var schema_1 = require("../schema");
exports.receivablesEnquiryPayIdResponseV1Schema = (0, schema_1.object)({
    durationMs: ['durationMs', (0, schema_1.optional)((0, schema_1.bigint)())],
    status: ['status', (0, schema_1.optional)((0, schema_1.string)())],
    statusDescription: ['statusDescription', (0, schema_1.optional)((0, schema_1.string)())],
    bankAccountNumber: ['BankAccountNumber', (0, schema_1.optional)((0, schema_1.bigint)())],
    lastResolutionDateTime: ['LastResolutionDateTime', (0, schema_1.optional)((0, schema_1.string)())],
    lastUpdatedDateTime: ['LastUpdatedDateTime', (0, schema_1.optional)((0, schema_1.string)())],
    payIdName: ['PayIdName', (0, schema_1.optional)((0, schema_1.string)())],
    registrationDateTime: ['RegistrationDateTime', (0, schema_1.optional)((0, schema_1.string)())]
});
