// sdksio-monoova-sdk/dist/cjs/models/rejectionReasonCodeEnum.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.rejectionReasonCodeEnumSchema = exports.RejectionReasonCodeEnum = void 0;
var schema_1 = require("../schema");
/**
 * Enum for RejectionReasonCodeEnum
 */
var RejectionReasonCodeEnum;
(function (RejectionReasonCodeEnum) {
    RejectionReasonCodeEnum["AM02"] = "AM02";
    RejectionReasonCodeEnum["AM06"] = "AM06";
    RejectionReasonCodeEnum["AM07"] = "AM07";
    RejectionReasonCodeEnum["BE05"] = "BE05";
    RejectionReasonCodeEnum["AG01"] = "AG01";
    RejectionReasonCodeEnum["MS01"] = "MS01";
    RejectionReasonCodeEnum["RR02"] = "RR02";
    RejectionReasonCodeEnum["RR03"] = "RR03";
    RejectionReasonCodeEnum["BE04"] = "BE04";
})(RejectionReasonCodeEnum = exports.RejectionReasonCodeEnum || (exports.RejectionReasonCodeEnum = {}));
/**
 * Schema for RejectionReasonCodeEnum
 */
exports.rejectionReasonCodeEnumSchema = (0, schema_1.stringEnum)(RejectionReasonCodeEnum);
