// sdksio-monoova-sdk/dist/cjs/models/value1Enum.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.value1EnumSchema = exports.Value1Enum = void 0;
var schema_1 = require("../schema");
/**
 * Enum for Value1Enum
 */
var Value1Enum;
(function (Value1Enum) {
    Value1Enum[Value1Enum["EnumNone"] = 0] = "EnumNone";
    Value1Enum[Value1Enum["EnumDaily"] = 1] = "EnumDaily";
    Value1Enum[Value1Enum["EnumBeginningOfMonth"] = 2] = "EnumBeginningOfMonth";
    Value1Enum[Value1Enum["EnumEndOfMonth"] = 3] = "EnumEndOfMonth";
    Value1Enum[Value1Enum["EnumMidMonth"] = 4] = "EnumMidMonth";
    Value1Enum[Value1Enum["EnumOnDemand"] = 5] = "EnumOnDemand";
    Value1Enum[Value1Enum["EnumOnDemandWithCatch"] = 6] = "EnumOnDemandWithCatch";
    Value1Enum[Value1Enum["EnumOncePerDay"] = 7] = "EnumOncePerDay";
    Value1Enum[Value1Enum["EnumIgnore"] = 8] = "EnumIgnore";
})(Value1Enum = exports.Value1Enum || (exports.Value1Enum = {}));
/**
 * Schema for Value1Enum
 */
exports.value1EnumSchema = (0, schema_1.numberEnum)(Value1Enum);
