// sdksio-monoova-sdk/dist/cjs/models/verifyInitiate2.js
"use strict";
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.verifyInitiate2Schema = void 0;
var schema_1 = require("../schema");
var creditMethodEnum_1 = require("./creditMethodEnum");
var payIdTypeEnum_1 = require("./payIdTypeEnum");
exports.verifyInitiate2Schema = (0, schema_1.object)({
    creditMethod: ['creditMethod', creditMethodEnum_1.creditMethodEnumSchema],
    bsb: ['bsb', (0, schema_1.optional)((0, schema_1.string)())],
    bankAccountNumber: ['bankAccountNumber', (0, schema_1.optional)((0, schema_1.number)())],
    accountName: ['accountName', (0, schema_1.string)()],
    remitter: ['remitter', (0, schema_1.optional)((0, schema_1.string)())],
    payIdType: ['payIdType', (0, schema_1.optional)(payIdTypeEnum_1.payIdTypeEnumSchema)],
    payId: ['payId', (0, schema_1.optional)((0, schema_1.string)())],
    verificationIdentifier: ['verificationIdentifier', (0, schema_1.string)()]
});
