// sdksio-monoova-sdk/dist/esm/models/disbursement1.js
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { lazy, number, object, optional, string } from '../schema';
import { disbursementMethod1EnumSchema, } from './disbursementMethod1Enum';
import { toBPayDetailsSchema } from './toBPayDetails';
import { toDirectCreditDetailsSchema, } from './toDirectCreditDetails';
import { toNppCreditBankAccountDetailsSchema, } from './toNppCreditBankAccountDetails';
import { toNppCreditPayIdDetailsSchema, } from './toNppCreditPayIdDetails';
export const disbursement1Schema = object({
    disbursementMethod: [
        'disbursementMethod',
        optional(disbursementMethod1EnumSchema),
    ],
    description: ['description', optional(string())],
    toMAccount: ['toMAccount', optional(string())],
    amount: ['amount', optional(number())],
    toMWallet: ['toMWallet', optional(string())],
    toDirectCreditDetails: [
        'toDirectCreditDetails',
        optional(lazy(() => toDirectCreditDetailsSchema)),
    ],
    lodgementReference: ['lodgementReference', optional(string())],
    toToken: ['toToken', optional(string())],
    toBPayDetails: ['toBPayDetails', optional(lazy(() => toBPayDetailsSchema))],
    toNppCreditBankAccountDetails: [
        'toNppCreditBankAccountDetails',
        optional(lazy(() => toNppCreditBankAccountDetailsSchema)),
    ],
    toNppCreditPayIdDetails: [
        'toNppCreditPayIdDetails',
        optional(lazy(() => toNppCreditPayIdDetailsSchema)),
    ],
});
