// sdksio-monoova-sdk/dist/esm/models/dTOBPAYBillerV1.js
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { boolean, number, object, optional, string } from '../schema';
export const dTOBPAYBillerV1Schema = object({
    billerCode: ['billerCode', optional(string())],
    billerLongName: ['billerLongName', optional(string())],
    billerShortName: ['billerShortName', optional(string())],
    acceptedPaymentMethods: ['acceptedPaymentMethods', optional(string())],
    activationDate: ['activationDate', optional(string())],
    deactivationDate: ['deactivationDate', optional(string())],
    minPaymentAmount: ['minPaymentAmount', optional(number())],
    maxPaymentAmount: ['maxPaymentAmount', optional(number())],
    crnValidationRuleName: ['crnValidationRuleName', optional(string())],
    checkDigitRuleName: ['checkDigitRuleName', optional(string())],
    lengthMask: ['lengthMask', optional(string())],
    fixedDigits: ['fixedDigits', optional(string())],
    isVariableCrn: ['isVariableCrn', optional(boolean())],
});
