// sdksio-monoova-sdk/dist/esm/models/dTOGenericPaymentGenericFinanceResponseV2.js
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { bigint, lazy, number, object, optional, string, } from '../schema';
import { dTOGenericPaymentFeeBreakdownSchema, } from './dTOGenericPaymentFeeBreakdown';
export const dTOGenericPaymentGenericFinanceResponseV2Schema = object({
    durationMs: ['durationMs', optional(bigint())],
    status: ['status', optional(string())],
    statusDescription: ['statusDescription', optional(string())],
    transactionId: ['transactionId', optional(number())],
    feeAmountExcludingGst: ['feeAmountExcludingGst', optional(number())],
    feeAmountGstComponent: ['feeAmountGstComponent', optional(number())],
    feeAmountIncludingGst: ['feeAmountIncludingGst', optional(number())],
    callerUniqueReference: ['callerUniqueReference', optional(string())],
    feeBreakdown: [
        'feeBreakdown',
        optional(lazy(() => dTOGenericPaymentFeeBreakdownSchema)),
    ],
});
