// sdksio-monoova-sdk/dist/esm/models/dTOGenericPaymentGatewayTransactionStatusResponseV2.js
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { bigint, object, optional, string } from '../schema';
import { transactionStatusEnumSchema, } from './transactionStatusEnum';
export const dTOGenericPaymentGatewayTransactionStatusResponseV2Schema = object({
    durationMs: ['durationMs', optional(bigint())],
    status: ['status', optional(string())],
    statusDescription: ['statusDescription', optional(string())],
    completedDate: ['completedDate', optional(string())],
    creditCardPaymentStatus: ['creditCardPaymentStatus', optional(string())],
    dishonouredDate: ['dishonouredDate', optional(string())],
    expectedClearanceDateForFunds: [
        'expectedClearanceDateForFunds',
        optional(string()),
    ],
    fundsClearedDate: ['fundsClearedDate', optional(string())],
    transactionStatus: [
        'transactionStatus',
        optional(transactionStatusEnumSchema),
    ],
    uniqueReference: ['uniqueReference', optional(string())],
});
