// sdksio-monoova-sdk/dist/esm/models/dTOGenericPaymentSettingsResponseV1.js
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { array, bigint, boolean, lazy, number, object, optional, string, } from '../schema';
import { dTOGenericPaymentDisbursementFeeV1Schema, } from './dTOGenericPaymentDisbursementFeeV1';
import { dTOGenericPaymentLoadFeeV1Schema, } from './dTOGenericPaymentLoadFeeV1';
export const dTOGenericPaymentSettingsResponseV1Schema = object({
    durationMs: ['durationMs', optional(bigint())],
    status: ['status', optional(string())],
    statusDescription: ['statusDescription', optional(string())],
    disbursementFees: [
        'disbursementFees',
        optional(array(lazy(() => dTOGenericPaymentDisbursementFeeV1Schema))),
    ],
    loadFees: [
        'loadFees',
        optional(array(lazy(() => dTOGenericPaymentLoadFeeV1Schema))),
    ],
    canDebitCreditCards: ['canDebitCreditCards', optional(boolean())],
    canPayBpay: ['canPayBpay', optional(boolean())],
    canImpersonate: ['canImpersonate', optional(boolean())],
    canDirectDebit: ['canDirectDebit', optional(boolean())],
    canDirectCredit: ['canDirectCredit', optional(boolean())],
    canCreateMAccounts: ['canCreateMAccounts', optional(boolean())],
    canCreateMWallets: ['canCreateMWallets', optional(boolean())],
    canAccessUserDatabase: ['canAccessUserDatabase', optional(boolean())],
    isIssuer: ['isIssuer', optional(boolean())],
    issuerMAccountNumber: ['issuerMAccountNumber', optional(string())],
    monthlyFeeExGst: ['monthlyFeeExGst', optional(number())],
    clearingMAccountNumber: ['clearingMAccountNumber', optional(string())],
    feeMAccountNumber: ['feeMAccountNumber', optional(string())],
    feeMAccountMonthlyFeeExGst: [
        'feeMAccountMonthlyFeeExGst',
        optional(number()),
    ],
    childMAccountSetUpFeeExGst: [
        'childMAccountSetUpFeeExGst',
        optional(number()),
    ],
    childMAccountMonthlyFeeExGst: [
        'childMAccountMonthlyFeeExGst',
        optional(number()),
    ],
    requiresClearedFundsOnly: ['requiresClearedFundsOnly', optional(boolean())],
});
