// sdksio-monoova-sdk/dist/esm/models/dTOMAccountTransactionLineItem.js
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { boolean, number, object, optional, string } from '../schema';
export const dTOMAccountTransactionLineItemSchema = object({
    credit: ['credit', optional(number())],
    dateTime: ['dateTime', optional(string())],
    debit: ['debit', optional(number())],
    description: ['description', optional(string())],
    destinationMAccountCardNumber: [
        'destinationMAccountCardNumber',
        optional(string()),
    ],
    isWaitingForFundsToClear: ['isWaitingForFundsToClear', optional(boolean())],
    mPaymentsId: ['mPaymentsId', optional(number())],
    mWalletAccountNumber: ['mWalletAccountNumber', optional(string())],
    rowOrder: ['rowOrder', optional(number())],
    runningBalance: ['runningBalance', optional(number())],
    sourceMAccountCardNumber: ['sourceMAccountCardNumber', optional(string())],
    subTransactionType: ['subTransactionType', optional(string())],
    transactionDescription: ['transactionDescription', optional(string())],
    transactionId: ['transactionId', optional(number())],
    transactionType: ['transactionType', optional(string())],
    uniqueReference: ['uniqueReference', optional(string())],
});
