// sdksio-monoova-sdk/dist/esm/models/dTOMAccountDetails.js
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { array, bigint, boolean, lazy, number, object, optional, string, } from '../schema';
import { dTOMAccountFinancialsSchema, } from './dTOMAccountFinancials';
import { dTOOptionItemSchema } from './dTOOptionItem';
export const dTOMAccountDetailsSchema = object({
    accountNumber: ['accountNumber', optional(bigint())],
    name: ['name', optional(string())],
    abn: ['abn', optional(string())],
    contactName: ['contactName', optional(string())],
    contactNumber: ['contactNumber', optional(string())],
    email: ['email', optional(string())],
    addressLine1: ['addressLine1', optional(string())],
    addressLine2: ['addressLine2', optional(string())],
    suburb: ['suburb', optional(string())],
    state: ['state', optional(string())],
    postCode: ['postCode', optional(string())],
    bsb: ['bsb', optional(string())],
    bankAccountNumber: ['bankAccountNumber', optional(number())],
    bankAccountTitle: ['bankAccountTitle', optional(string())],
    allowDuplicates: ['allowDuplicates', optional(boolean())],
    feeAccount: ['feeAccount', optional(number())],
    financials: ['financials', optional(lazy(() => dTOMAccountFinancialsSchema))],
    options: ['options', optional(array(lazy(() => dTOOptionItemSchema)))],
});
