// sdksio-monoova-sdk/dist/esm/models/dTOMAccountFinancials.js
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { boolean, lazy, number, object, optional, string, } from '../schema';
import { dTOMAccountClearedFundsAccountSchema, } from './dTOMAccountClearedFundsAccount';
export const dTOMAccountFinancialsSchema = object({
    accountNumber: ['accountNumber', optional(number())],
    actualBalance: ['actualBalance', optional(number())],
    clearedFundsOnly: ['clearedFundsOnly', optional(boolean())],
    creditLimit: ['creditLimit', optional(number())],
    feeAccountActualBalance: ['feeAccountActualBalance', optional(number())],
    feeAccountNumber: ['feeAccountNumber', optional(number())],
    hasSeparateFeeAccount: ['hasSeparateFeeAccount', optional(boolean())],
    lowerCreditLimit: ['lowerCreditLimit', optional(number())],
    minimumSettlementAmount: ['minimumSettlementAmount', optional(number())],
    settlementFrequency: ['settlementFrequency', optional(string())],
    upperCreditLimit: ['upperCreditLimit', optional(number())],
    isFeeAccount: ['isFeeAccount', optional(boolean())],
    isParentAccount: ['isParentAccount', optional(boolean())],
    isParentClearedFunds: ['isParentClearedFunds', optional(boolean())],
    clearedFundsAccountFinancials: [
        'clearedFundsAccountFinancials',
        optional(lazy(() => dTOMAccountClearedFundsAccountSchema)),
    ],
});
