// sdksio-monoova-sdk/dist/esm/models/dTOMWalletDetails.js
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { array, boolean, lazy, object, optional, string, } from '../schema';
import { dTOMWalletFinancialsSchema, } from './dTOMWalletFinancials';
import { dTOMWalletOptionItemSchema, } from './dTOMWalletOptionItem';
export const dTOMWalletDetailsSchema = object({
    createdDateTime: ['createdDateTime', optional(string())],
    accountNumber: ['accountNumber', optional(string())],
    name: ['name', optional(string())],
    nickName: ['nickName', optional(string())],
    email: ['email', optional(string())],
    mobile: ['mobile', optional(string())],
    lastTransactionDate: ['lastTransactionDate', optional(string())],
    contactPhone: ['contactPhone', optional(string())],
    authenticationEmail: ['authenticationEmail', optional(string())],
    authenticationMobile: ['authenticationMobile', optional(string())],
    authenticationLandLine: ['authenticationLandLine', optional(string())],
    registeredByMAccountNumber: [
        'registeredByMAccountNumber',
        optional(string()),
    ],
    isOnHold: ['isOnHold', optional(boolean())],
    isClosed: ['isClosed', optional(boolean())],
    isStolen: ['isStolen', optional(boolean())],
    isRegistered: ['isRegistered', optional(boolean())],
    financials: ['financials', optional(lazy(() => dTOMWalletFinancialsSchema))],
    options: ['options', optional(array(lazy(() => dTOMWalletOptionItemSchema)))],
});
