// sdksio-monoova-sdk/dist/esm/models/dTOToolsAbnValidateResponseV1.js
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { bigint, boolean, object, optional, string } from '../schema';
export const dTOToolsAbnValidateResponseV1Schema = object({
    durationMs: ['durationMs', optional(bigint())],
    status: ['status', optional(string())],
    statusDescription: ['statusDescription', optional(string())],
    abn: ['abn', optional(string())],
    lastUpdatedDate: ['lastUpdatedDate', optional(string())],
    isAbnValid: ['isAbnValid', optional(boolean())],
    isCharity: ['isCharity', optional(boolean())],
    entityTypeCode: ['entityTypeCode', optional(string())],
    entityTypeDescription: ['entityTypeDescription', optional(string())],
    organisationName: ['organisationName', optional(string())],
    legalName: ['legalName', optional(string())],
    organisationNameFromDate: ['organisationNameFromDate', optional(string())],
    businessPhysicalAddressStateCode: [
        'businessPhysicalAddressStateCode',
        optional(string()),
    ],
    businessPhysicalAddressPostCode: [
        'businessPhysicalAddressPostCode',
        optional(string()),
    ],
    businessPhysicalAddressEffectiveFromDate: [
        'businessPhysicalAddressEffectiveFromDate',
        optional(string()),
    ],
});
