// sdksio-monoova-sdk/dist/esm/models/dTOVerifyAccountDetails.js
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { boolean, number, object, optional, string } from '../schema';
export const dTOVerifyAccountDetailsSchema = object({
    token: ['token', optional(string())],
    created: ['created', optional(string())],
    updated: ['updated', optional(string())],
    lastSystemBSBValidation: ['lastSystemBSBValidation', optional(string())],
    comment: ['comment', optional(string())],
    bsb: ['bsb', optional(string())],
    bankAccountNumber: ['bankAccountNumber', optional(number())],
    bankAccountTitle: ['bankAccountTitle', optional(string())],
    remitter: ['remitter', optional(string())],
    isBlacklisted: ['isBlacklisted', optional(boolean())],
    hasDDAuthority: ['hasDDAuthority', optional(boolean())],
    ddAuthorityChangedTimestamp: [
        'ddAuthorityChangedTimestamp',
        optional(string()),
    ],
    ddAuthorityChangedBy: ['ddAuthorityChangedBy', optional(string())],
    verificationIdentifier: ['verificationIdentifier', optional(string())],
    verificationStatus: ['verificationStatus', optional(string())],
    verificationSignonMerchant: [
        'verificationSignonMerchant',
        optional(string()),
    ],
    verificationRetryCount: ['verificationRetryCount', optional(number())],
    verificationRetyCountBeforeLockout: [
        'verificationRetyCountBeforeLockout',
        optional(number()),
    ],
    verificationLastRetryTimestamp: [
        'verificationLastRetryTimestamp',
        optional(string()),
    ],
    verificationLastRetryLockoutMinutes: [
        'verificationLastRetryLockoutMinutes',
        optional(number()),
    ],
    verificationDEJobId: ['verificationDEJobId', optional(number())],
    verificationRootId: ['verificationRootId', optional(number())],
});
