// sdksio-monoova-sdk/dist/esm/models/eventNppReturnWebhook.js
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { number, object, optional, string } from '../schema';
import { payIdType1EnumSchema } from './payIdType1Enum';
export const eventNppReturnWebhookSchema = object({
    returnDate: ['ReturnDate', optional(string())],
    amount: ['Amount', optional(number())],
    bsb: ['Bsb', optional(string())],
    accountNumber: ['AccountNumber', optional(string())],
    accountName: ['AccountName', optional(string())],
    payId: ['payId', optional(string())],
    payIdType: ['payIdType', optional(payIdType1EnumSchema)],
    endToEndId: ['EndToEndId', optional(string())],
    returnReason: ['ReturnReason', optional(string())],
    transactionDate: ['TransactionDate', optional(string())],
    transactionId: ['TransactionId', optional(number())],
    uniqueReference: ['UniqueReference', optional(string())],
    remitterName: ['RemitterName', optional(string())],
    lodgementRef: ['LodgementRef', optional(string())],
    returnTransactionId: ['ReturnTransactionId', optional(number())],
});
