// sdksio-monoova-sdk/dist/esm/models/eventNppPaymentStatusWebhook.js
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { number, object, optional, string } from '../schema';
import { payIdType1EnumSchema } from './payIdType1Enum';
export const eventNppPaymentStatusWebhookSchema = object({
    transactionDate: ['TransactionDate', optional(string())],
    uniqueReference: ['UniqueReference', optional(string())],
    transactionId: ['TransactionId', optional(number())],
    status: ['Status', optional(string())],
    amount: ['Amount', optional(number())],
    bsb: ['Bsb', optional(string())],
    accountNumber: ['AccountNumber', optional(string())],
    accountName: ['AccountName', optional(string())],
    payId: ['PayId', optional(string())],
    payIdType: ['PayIdType', optional(payIdType1EnumSchema)],
    endToEndId: ['EndToEndId', optional(string())],
    remitterName: ['RemitterName', optional(string())],
    lodgementRef: ['LodgementRef', optional(string())],
    rejectionDateTime: ['RejectionDateTime', optional(string())],
    rejectionReasonDescription: [
        'RejectionReasonDescription',
        optional(string()),
    ],
    rejectionTransactionId: ['RejectionTransactionId', optional(number())],
});
