// sdksio-monoova-sdk/dist/esm/models/eventPayToreceivePaymentWebhook.js
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { object, optional, string } from '../schema';
export const eventPayToreceivePaymentWebhookSchema = object({
    transactionId: ['TransactionId', optional(string())],
    dateTime: ['DateTime', optional(string())],
    remitterName: ['RemitterName', optional(string())],
    amount: ['Amount', optional(string())],
    accountName: ['AccountName', optional(string())],
    accountNumber: ['AccountNumber', optional(string())],
    bsb: ['Bsb', optional(string())],
    paymentDescription: ['PaymentDescription', optional(string())],
    payId: ['payId', optional(string())],
    payIdName: ['payIdName', optional(string())],
    sourceBsb: ['SourceBsb', optional(string())],
    sourceAccountNumber: ['SourceAccountNumber', optional(string())],
    sourceAccountName: ['SourceAccountName', optional(string())],
    endToEndId: ['EndToEndId', optional(string())],
    categoryPurposeCode: ['CategoryPurposeCode', optional(string())],
    creditorReferenceInformation: [
        'CreditorReferenceInformation',
        optional(string()),
    ],
    uSINumber: ['USINumber', optional(string())],
    uSIScheme: ['USIScheme', optional(string())],
    ultimateCreditorName: ['UltimateCreditorName', optional(string())],
    mmsId: ['mmsId', optional(string())],
    initiatingPartyName: ['InitiatingPartyName', optional(string())],
});
