// sdksio-monoova-sdk/dist/esm/models/inboundDirectDebitWebhookDirectDebitDetails.js
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { number, object, optional, string } from '../schema';
export const inboundDirectDebitWebhookDirectDebitDetailsSchema = object({
    transactionId: ['TransactionId', optional(string())],
    batchId: ['BatchId', optional(string())],
    dateTime: ['DateTime', optional(string())],
    status: ['Status', optional(string())],
    respondBeforeDateTime: ['RespondBeforeDateTime', optional(string())],
    bsb: ['Bsb', optional(string())],
    accountNumber: ['AccountNumber', optional(number())],
    accountName: ['AccountName', optional(string())],
    amount: ['Amount', optional(number())],
    lodgementRef: ['LodgementRef', optional(string())],
    remitterName: ['RemitterName', optional(string())],
    nameOfUserSupplyingFile: ['NameOfUserSupplyingFile', optional(string())],
    numberOfUserSupplyingFile: [
        'NumberOfUserSupplyingFile',
        optional(string()),
    ],
    descriptionOfEntriesOnFile: [
        'DescriptionOfEntriesOnFile',
        optional(string()),
    ],
    indicator: ['Indicator', optional(string())],
    withholdingTaxAmount: ['WithholdingTaxAmount', optional(number())],
    sourceBsb: ['SourceBsb', optional(string())],
    sourceAccountNumber: ['SourceAccountNumber', optional(string())],
});
