// sdksio-monoova-sdk/dist/esm/models/mAccountUpdate.js
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { array, boolean, lazy, number, object, optional, string, } from '../schema';
import { optionSchema } from './option';
export const mAccountUpdateSchema = object({
    accountNumber: ['accountNumber', string()],
    name: ['name', optional(string())],
    abn: ['abn', string()],
    contactName: ['contactName', string()],
    contactNumber: ['contactNumber', number()],
    email: ['email', string()],
    addressLine1: ['addressLine1', optional(string())],
    addressLine2: ['addressLine2', string()],
    suburb: ['suburb', string()],
    state: ['state', string()],
    postCode: ['postCode', optional(string())],
    bsb: ['bsb', optional(string())],
    bankAccountNumber: ['bankAccountNumber', optional(number())],
    bankAccountTitle: ['bankAccountTitle', optional(string())],
    financials: ['financials', optional(boolean())],
    options: ['options', optional(array(lazy(() => optionSchema)))],
});
