// sdksio-monoova-sdk/dist/esm/models/pendingRtgsImtWebhookDetails.js
/**
 * Monoova Payments APILib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { number, object, optional, string } from '../schema';
export const pendingRtgsImtWebhookDetailsSchema = object({
    transactionId: ['TransactionId', optional(string())],
    accountNumber: ['AccountNumber', optional(string())],
    bSB: ['BSB', optional(string())],
    accountName: ['AccountName', optional(string())],
    transactionReferenceNumber: [
        'TransactionReferenceNumber',
        optional(string()),
    ],
    amount: ['Amount', optional(number())],
    status: ['Status', optional(string())],
    lodgementRef: ['LodgementRef', optional(string())],
    date: ['Date', optional(string())],
    sourceDetails: ['SourceDetails', optional(string())],
    respondBeforeDateTime: ['RespondBeforeDateTime', optional(string())],
    sendingOrganisation: ['SendingOrganisation', optional(string())],
    receivingOrganisation: ['ReceivingOrganisation', optional(string())],
    orderingInstitution: ['OrderingInstitution', optional(string())],
    originatingAddress: ['OriginatingAddress', optional(string())],
});
